src/test_edgepi/unit_tests/test_adc/test_adc_voltage.py

```diff
@@ -22,27 +27,89 @@
 GAIN = 1
 OFFSET = 0
 
+@pytest.mark.parametrize("code, result",
+                        [([0xFF,0xFF,0xFF,0xFF], True),
+                         ([0x7F,0xFF,0xFF,0xFF], False),
+                        ])
+def test_is_negative_voltage(code, result):
+    code_bits = bitstring_from_list(code)
+    assert _is_negative_voltage(code_bits) ==result
+
 
 @pytest.mark.parametrize(
     "code, voltage, num_bytes",
     [
-        (0x00000000, 0, ADCNum.ADC_1.value.num_data_bytes),
-        (0x00000000, 0, ADCNum.ADC_2.value.num_data_bytes),
+        ([0,0,0,0], 0, ADCNum.ADC_1.value.num_data_bytes),
+        ([0,0,0,0], 0, ADCNum.ADC_2.value.num_data_bytes),
         # based on a reference voltage of 2.5 V
         (
-            0xFFFFFFFF,
-            5.0,
+            [0x7F,0xFF,0xFF,0xFF],
+            2.5,
+            ADCNum.ADC_1.value.num_data_bytes,
+        ),        (
+            [0x80,0x0,0x0,0x0],
+            -2.5,
             ADCNum.ADC_1.value.num_data_bytes,
         ),
         (
-            0xFFFFFF,
-            5.0,
+            [0x7F,0xFF,0xFF,0xFF],
+            2.5,
+            ADCNum.ADC_2.value.num_data_bytes,
+        ),
+        (
+            [0x80,0x0,0x0,0x0],
+            -2.5,
+            ADCNum.ADC_2.value.num_data_bytes,
+        ),
+    ],
+)
+def test_code_to_input_voltage(code, voltage, num_bytes):
```

Comment:
can you also add tests for max and min supported values?

Reply:
[0x7F,0xFF,0xFF,0xFF] is the max and [0x80,0x0,0x0,0x0] is the min value.